components/blocks/testimonialsList.tsx

```diff
@@ -0,0 +1,111 @@
+import Image from "next/image";
+import { useEffect, useState } from "react";
+import { FaSpinner } from "react-icons/fa";
+import type { Template } from "tinacms";
+import { TinaMarkdown } from "tinacms/dist/rich-text";
+import client from "../../.tina/__generated__/client";
+import { Rating } from "../util/consulting/rating";
+
+const defaultAvatar = "/images/thumbs/avatar-thumbnail.png";
+
+/**
+ * Render a table of newsletters.
+ * @param data The data for the table.
+ * @returns The table component.
+ */
+export const TestimonialsList = ({ data: { hideInternshipTestimonials } }) => {
+  const [testimonials, setTestimonials] = useState([]);
+  const [hasLoaded, setHasLoaded] = useState(false);
+
+  useEffect(() => {
+    if (!hasLoaded) {
+      loadTestimonials();
+    }
+  }, [hasLoaded]);
+
+  const loadTestimonials = () => {
+    client.queries.testimonialsConnection().then((data) => {
+      const testimonials = data.data?.testimonialsConnection?.edges?.map(
+        (edge) => edge?.node
+      );
+
+      const sortedTestimonials = testimonials
+        ?.filter(
+          (testimonial) =>
+            testimonial?.categories !== null &&
+            testimonial?.categories[0]?.category.name !== "Internship"
+        )
+        ?.map((testimonial) => testimonial);
+
+      if (hideInternshipTestimonials) {
+        setTestimonials(sortedTestimonials);
+      } else {
+        setTestimonials(testimonials);
+      }
+      setHasLoaded(true);
+    });
+  };
```

Comment:
If we are loading the testimonials in `getStaticProps` anyway, why do we need to do a client-side fetch of testimonial data?